File: scalechain-proto-codec/src/test/scala/io/scalechain/blockchain/proto/codec/walletparts/OwnershipDescriptorCodecSpec.scala
package io.scalechain.blockchain.proto.codec.walletparts

import io.scalechain.blockchain.proto.{WalletOutput, Hash, OwnershipDescriptor}
import io.scalechain.blockchain.proto.codec._
import io.scalechain.blockchain.proto.codec.primitive.CodecSuite
import io.scalechain.blockchain.proto.test.ProtoTestData

class OwnershipDescriptorCodecSpec extends CodecSuite with ProtoTestData {

  implicit val ownershipDescriptorCodec = OwnershipDescriptorCodec.codec

  val ownershipDescriptor = OwnershipDescriptor(
    account  = "kangmo",
    privateKeys = List( "00000000bd0ed80435fc9fe3269da69bb0730ebb454d0a29128a870ea1a37929" )
  )

  "OwnershipDescriptorCodec" should {

    "roundtrip" in {
      roundtrip(ownershipDescriptor)
    }
/*
    "test" in {

      val serialized = OwnershipDescriptorCodec.serialize( ownershipDescriptor )
      val parsed : OwnershipDescriptor = OwnershipDescriptorCodec.parse(serialized)
      parsed shouldBe ownershipDescriptor
    }
    */
  }
}

File: scalechain-proto-codec/src/test/kotlin/io/scalechain/blockchain/proto/codec/walletparts/OwnershipDescriptorCodecSpec.kt
package io.scalechain.blockchain.proto.codec.walletparts

import io.kotlintest.KTestJUnitRunner
import io.kotlintest.matchers.Matchers
import io.kotlintest.specs.FlatSpec
import io.scalechain.blockchain.proto.*
import io.scalechain.blockchain.proto.codec.CodecTestUtil
import io.scalechain.blockchain.proto.codec.*
import org.junit.runner.RunWith


@RunWith(KTestJUnitRunner::class)
class OwnershipDescriptorCodecSpec : FlatSpec(), Matchers, CodecTestUtil {


  val ownershipDescriptor = OwnershipDescriptor(
    account  = "kangmo",
    privateKeys = listOf( "00000000bd0ed80435fc9fe3269da69bb0730ebb454d0a29128a870ea1a37929" )
  )

  init {

    "OwnershipDescriptorCodec" should "roundtrip" {
      roundTrip(OwnershipDescriptorCodec, ownershipDescriptor) shouldBe true
    }
  }
/*
    "test" {

      val serialized = OwnershipDescriptorCodec.serialize( ownershipDescriptor )
      val parsed : OwnershipDescriptor = OwnershipDescriptorCodec.parse(serialized)
      parsed shouldBe ownershipDescriptor
    }
    */
}

